app/views/unattended/provisioning_templates/provision/coreos_provision.erb

```diff
@@ -1,46 +1,56 @@
-#cloud-config
 <%#
 kind: provision
 name: CoreOS provision
 model: ProvisioningTemplate
 oses:
 - CoreOS
+- RedHatCoreOS
+- FedoraCoreOS
+- Flatcar
 %>
-<%
-os_major  = @host.operatingsystem.major.to_i
-%>
-coreos:
+<%- if host_param('ignition_template') -%>
+<%= host_param('ignition_template') %>
+<%- else -%>
+<%- config = <<EOS
+---
+passwd:
+  #{indent(2) { snippet_if_exists('CoreOS provision custom passwd') }}
+  users:
+    - name: core
+      passwd: #{root_pass}
+      ssh_authorized_keys: #{@host.ssh_authorized_keys}
+systemd:
+  #{indent(2) { snippet_if_exists('CoreOS provision custom systemd') }}
   units:
-    - name: coreos-bootstrap.service
-      runtime: no
-      command: start
-      content: |
+    - name: os-install.service
+      enabled: true
+      contents: |
         [Unit]
-        Description=Install coreos to disk
+        Description=Install to disk
+        Requires=network-online.target
+        After=network-online.target
         [Service]
         Type=oneshot
-        ExecStart=/usr/bin/coreos-install \
-          -C <%= @host.operatingsystem.release_name %> \
-          -V <%= @host.operatingsystem.release %> \
-          -d <%= host_param('install-disk') || '/dev/sda' %> <% if @host.provider == 'VMware' %>-o vmware_raw<% end %> \
-          -c /home/core/cloud-config.yml <% if os_major >= 557 -%>-b <%= @mediapath %><% end %>
-        ExecStartPost=/usr/bin/wget -q -O /dev/null --no-check-certificate <%= foreman_url('built') %>
-        ExecStartPost=/usr/sbin/reboot
-        [X-Fleet]
-        X-Conflicts=coreos-bootstrap.service
-<% if host_param('ssh_authorized_keys') -%>
-ssh_authorized_keys:
-<% host_param('ssh_authorized_keys').split(',').map{ |item| item.strip }.each do |ssh_key| -%>
-  - "<%= ssh_key %>"
-<% end -%>
-<% else -%>
-users:
-  - name: core
-    passwd: <%= root_pass %>
-<% end -%>
-write_files:
-  - content: |
-      <%= snippet 'coreos_cloudconfig' %>
-    path: /home/core/cloud-config.yml
-    permissions: '0600'
-    owner: core:core
+        ExecStart=/usr/bin/flatcar-install -v \
```

Comment:
I think this will fail for CoreOS/RedHatCoreOS/FedoraCoreOS because the installer has a different name for flatcar.

Reply:
This template is only meant for legacy CoreOS (thus Flatcar or other forks). But granted this is a leftover. I wonder how I should handle this. Maybe an if-statement, if OS name is Flatcar then use this, otherwise fallback to coreos-install?